cadc-vos-server/src/main/java/ca/nrc/cadc/vos/server/LocalServiceURI.java

```diff
@@ -70,44 +70,38 @@
 package ca.nrc.cadc.vos.server;
 
 
+import ca.nrc.cadc.vos.VOSURI;
+import ca.nrc.cadc.vos.server.util.BeanUtil;
+
 import java.net.URI;
 import java.net.URL;
 import java.util.Properties;
 
 import org.apache.log4j.Logger;
 
 /**
- * Class that simply tells the service what the currently running
- * serviceID is.  This information is retrieved from the file
- * LocalServiceURI.properties read from the classpath.  The format
- * of the file should be, for example:
- *
- * serviceURI = ivo://cadc.nrc.ca/vospace
+ * Class used to retrieve the serviceURI of the running vospace instance. 
  *
  */
 public class LocalServiceURI
 {
     private static final Logger log = Logger.getLogger(LocalServiceURI.class);
 
-    private static final String LOCAL_PROP_FILE = LocalServiceURI.class.getSimpleName() + ".properties";
-    private static final String URI_KEY = "serviceURI";
-
     private URI serviceURI;
 
     public LocalServiceURI()
     {
-        try
-        {
-            URL propFile = LocalServiceURI.class.getClassLoader().getResource(LOCAL_PROP_FILE);
-            Properties props = new Properties();
-            props.load(propFile.openStream());
-            String uriString = props.getProperty(URI_KEY);
-            this.serviceURI = new URI(uriString);
-        }
-        catch (Exception e)
-        {
-            log.warn("Unable to read local service URI from " + LOCAL_PROP_FILE +
-                    "\n    Reason: " + e.getMessage(), e);
+        try {
+            String uriBase = System.getProperty(BeanUtil.VOS_BASE_URI_PROP);
```

Comment:
I agree that a special LocalServiceURI.properties is cumbersome... but this self identity should really come from a config file and be a mandatory config item, so this change is going in the wrong direction. 

We'll have to undo this later when we get to putting all vospace services into containers... so I'm not really saying this needs to be changed now, but it is the wrong direction (experience with SI).

**However, there definitely should not be a default value to one service. That's just wrong and encourages incorrect config so that should be fixed.**

**Secondarily, if you are going to do a blanket catch, don't catch Throwable, catch Exception.** Throwable is also the parent of Error (eg JVM issues like out-of-memory) so you don't want to catch those. It's also bad to catch, log, and then return like nothing happened... but maybe that can't be fixed so easily in code we want to delete anyway.

Reply:
Okay, I was hesitant about making vault the default, so I'll change it to be a required config item.  I though however that in storage-inventory a lot of java config was moving to catalina.properties, thus the change to a system property.  I think we should get this right now though, so I'll change it back to a file if you think that's the best fit.

Reply:
in SI we're using catalina.properties for things that tomcat has to have values for (essentially, things in server.xml or context.xml) plus some generic/legacy stuff like registry override or that prop to trust client cert in a header (which is library config) but not in general... applications have their own config file and some libs still have a config file as well. This is a lib config file, so probably the right usage for containers.